src/main/webapp/app/pages/curation/mutation/MutationsSection.tsx

```diff
@@ -12,25 +12,47 @@ import { componentInject } from 'app/shared/util/typed-inject';
 import { IRootStore } from 'app/stores';
 import { observer } from 'mobx-react';
 import { Mutation } from 'app/shared/model/firebase/firebase.model';
-import { compareMutations } from 'app/shared/util/firebase/firebase-utils';
+import { compareMutations, getFirebaseGenePath } from 'app/shared/util/firebase/firebase-utils';
 import MutationCollapsible from '../collapsible/MutationCollapsible';
 import styles from '../styles.module.scss';
+import { onValue, ref } from 'firebase/database';
+import { Alteration, AnnotateAlterationBody } from 'app/shared/api/generated';
+import { REFERENCE_GENOME } from 'app/config/constants/constants';
+import { flow, flowResult } from 'mobx';
+import { AnnotatedAltsCache } from 'app/stores/curation-page.store';
 
 export interface IMutationsSectionProps extends StoreProps {
   mutationsPath: string;
   hugoSymbol: string;
   isGermline: boolean;
   parsedHistoryList: Map<string, ParsedHistoryRecord[]>;
 }
-
-function MutationsSection({ mutationsPath, hugoSymbol, isGermline, parsedHistoryList, addMutation }: IMutationsSectionProps) {
+function MutationsSection({
+  mutationsPath,
+  hugoSymbol,
+  isGermline,
+  parsedHistoryList,
+  addMutation,
+  firebaseDb,
+  annotatedAltsCache,
+}: IMutationsSectionProps) {
   const [showAddMutationModal, setShowAddMutationModal] = useState(false);
   const [filteredIndices, setFilteredIndices] = useState<number[]>([]);
   const [openMutationCollapsibleIndex, setOpenMutationCollapsibleIndex] = useState<number>(null);
 
   const mutationSectionRef = useRef<HTMLDivElement>(null);
   const mutationScrollContainerRef = useRef<HTMLDivElement>(null);
 
+  useEffect(() => {
+    onValue(
+      ref(firebaseDb, `${getFirebaseGenePath(isGermline, hugoSymbol)}/mutations`),
+      snapshot => {
+        annotatedAltsCache.fetch(hugoSymbol, snapshot.val());
```

Comment:
`fetch` takes `hugoSymbol` and `MutationQuery[]`, but `snapshot.val()` here is actually of type `Mutation[]`.